src/ServiceManagement/Compute/Commands.ServiceManagement/IaaS/Extensions/Chef/SetAzureVMChefExtension.cs

```diff
@@ -66,6 +68,18 @@ public class SetAzureVMChefExtensionCommand : VirtualMachineChefExtensionCmdletB
         public string RunList { get; set; }
 
         [Parameter(
+             ValueFromPipelineByPropertyName = true,
+             HelpMessage = "A JSON string to be added to the first run of chef-client. e.g. -JsonAttribute '{\"foo\" : \"bar\"}'")]
```

Comment:
@markcowl I have created the issue #3146 since there is no Automated test written for the Set;AzureVMChefExtension module and I have to write it from scratch. Also I have a dependency on #2173 for adding automated test in Resource management i.e. for Set-AzureRMVMChefExtension so I have created separate issue #3147 for this.